lib/middleware/websocket/interactors/client_interactor.rb

```diff
@@ -0,0 +1,30 @@
+require './lib/middleware/websocket/client'
+require './lib/middleware/websocket/interactors/update_all_clients'
+
+module Websocket
+  module Interactor
+    class ClientInteractor
+      attr_reader :clients
+
+      def initialize
+        @clients = []
+      end
+
+      def create_client(incoming_client: client)
+        @clients << Client.new(connection_client: incoming_client)
+      end
+
+      def delete_client(incoming_client: client)
+        @clients -= find_client(incoming_client: incoming_client)
+      end
+
+      def find_client(incoming_client: client)
+        @clients.select { |client| client.connection_client == incoming_client }
+      end
+
+      def update_all_clients
+        UpdateAllClients.new(clients: @clients)
+      end
+    end
+  end
+end
```

Comment:
While this class is very nice and neat, I want to challenge you on whether or not we strictly need it. Abstraction can be a great tool, but in this case we might be going further than we need to.

For the most part, this object can be summarized as an array that we can manipulate, which leads me to wonder if we'd be better of going with the "simpler" approach of using only an array.

Reply:
Updated in #25 .